kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -337,7 +337,12 @@ def get_path(self) -> Path:
         return self._env_path
 
     def install(self, settings: PythonSettings) -> None:
-        _run_with_uv_indexes(["uv", "sync", "--all-extras"], settings, self.project_directory)
+        # --locked is used only when uv.lock is present
+        _run_with_uv_indexes(
+            ["uv", "sync"] + (["--locked"] if (self.project_directory / "uv.lock").is_file() else []),
```

Comment:
personnal preference:
```suggestion
        # --locked is used only when uv.lock is present
        cmd = ["uv", "sync"] 
        if (self.project_directory / "uv.lock").is_file()
            cmd.append("--locked")
        _run_with_uv_indexes(
           cmd,
```
reformatting might be needed